File: ImplicIntent/app/src/main/java/com/example/implicintent/MainActivity.java
package com.example.implicintent;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    Button mbutton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mbutton=(Button)findViewById(R.id.button);



    }

    public void onClickOpenWebpageButton(View view){
        String urlAsString="http://www.udacity.com";
        openWebPage(urlAsString);
    }
    public void openWebPage(String url){
        Uri webPage=Uri.parse(url);
        Intent intent=new Intent(Intent.ACTION_VIEW,webPage);
        if(intent.resolveActivity(getPackageManager()) !=null){
            startActivity(intent);
        }
    }
}

File: Mapp/app/src/main/java/com/example/mapp/MainActivity.kt
package com.example.mapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollbutton: button =findViewById(R.id.roll_button)
        rollbutton.text ="lets roll"
    }
}
